File: js/euler19.js
function euler19(){
    var res = 0;
    for (let year = 1901; year <= 2000; year++){
        for (let month = 1; month <= 12; month++){
            let components = [year, month, 1];
            let first = new Date(components.join("-"));
            // Date.getDay() returns day of week
            // Sunday is first day of week in JS (0)
            if (first.getDay() == 0)  res++;
        }
    }
    return res;
}

console.log(euler19());

File: go/euler19.go
package main
import (
    "fmt"
    "time"
)

func main() {
    date := time.Date(1901, 1, 1, 0, 0, 0, 0, time.UTC) 
    res := 0
    // check all months in years from 1901 to 2000, inclusive
    for date.Year() < 2001 {
        // increment count if first day of month was a Sunday
        if int(date.Weekday()) == 0 {
            res++
        }
        // add one month to date
        date = date.AddDate(0, 1, 0)
    }
    fmt.Println(res)
}
